frontends/web/src/components/alert/Alert.tsx

```diff
@@ -31,79 +14,54 @@ import { Button } from '../forms';
 let alertUser: (message: string, options?: AlertUserOptions) => void;
 
 type AlertUserOptions = {
-    callback?: () => void,
-    asDialog?: boolean,
+  callback?: () => void;
+  asDialog?: boolean;
 };
 
-interface State {
-    active: boolean;
-    asDialog: boolean;
-    message?: string;
-}
+const Alert = () => {
+  const [active, setActive] = useState(false);
+  const [asDialog, setAsDialog] = useState(true);
+  const [message, setMessage] = useState<string>();
+  const [callback, setCallback] = useState<AlertUserOptions['callback']>();
+  const { t } = useTranslation();
 
-class Alert extends Component<WithTranslation, State> {
-  private callback?: () => void; // Assigned when alertUser is called / Called before close.
-
-  constructor(props: WithTranslation) {
-    super(props);
-    alertUser = this.alertUser;
-    this.state = {
-      active: false,
-      asDialog: true,
-    };
-  }
-
-  private handleClose = () => {
-    if (this.callback) {
-      this.callback();
-    }
-    this.setState({
-      active: false,
-    });
-  };
-
-  private alertUser = (
-    message: string,
-    options: AlertUserOptions = {},
-  ) => {
+  alertUser = (message: string, options: AlertUserOptions = {}) => {
     const {
-      callback,
       asDialog = true,
     } = options;
-    this.callback = callback;
-    this.setState({
-      active: true,
-      asDialog,
-      message,
-    });
+    setCallback(() => options.callback);
+    setActive(true);
+    setAsDialog(asDialog);
+    setMessage(message);
   };
 
-  public render() {
-    const { t } = this.props;
-    const { active, asDialog, message } = this.state;
-    return (active && message) ? (
-      <form onSubmit={this.handleClose}>
-        <View
-          key="alert-overlay"
-          dialog={asDialog}
-          textCenter={!asDialog}
-          fullscreen>
-          <ViewHeader title={<MultilineMarkup tagName="span" markup={message} />} />
-          <ViewButtons>
-            <Button
-              autoFocus
-              primary
-              onClick={this.handleClose}>
-              {t('button.ok')}
-            </Button>
-          </ViewButtons>
-        </View>
-      </form>
-    ) : null;
-  }
-}
+  const handleClose = () => {
+    if (callback) {
+      callback();
+    }
+    setActive(false);
+  };
 
-const TranslatedAlert = withTranslation()(Alert);
+  return active && message ? (
```

Comment:
Nit: for readability it would help me read the code if the condition before conditional (ternary) operator was in parentheses